code-challenges.test.js

```diff
@@ -16,58 +19,119 @@
-
-
-
-
-
-// --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
-
-// a) Create a test with expect statements for each of the variables provided.
-
-var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
-
-var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
-// Expected output: [-823, 7, 23]
-
-
-
-
-// b) Create the function that makes the test pass.
-
-
-
-
-
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
-
-var numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
-
-var numbersToAdd3 = []
-// Expected output: []
-
-
-
-
-
-// b) Create the function that makes the test pass.
+// code describe with the name of the function
+// write variables for array input and outputs
+// test what you want the the function to do.
+// write expect statement .
+fdescribe("fibLength", () => {
+    let input = ["6"];
+    let outPut = [1, 1, 2, 3, 5, 8];
+    test ("return a new array of that length containing the numbers of the Fibonacci", () => {
+      expect(fibLength(input)).toEqual(outPut)
+      });
+  });
+  
+  
+  // Example input: 6
+  // Expected output: [1, 1, 2, 3, 5, 8]
+  
+  // Example input: 10
+  // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+  
+  
+  
+  
+  
+  // b) Create the function that makes the test pass.
+  
+  var output1 = [1, 1, 2, 3, 5, 8]
+  var outPut2 = [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+  
+  const fibLength = (array) => {
+  return array.map(value(outPut1, outPut2))
+  }
+  
+  
+ 
+  
+  
+  
+  // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+  
+  //use filter or if the number is not divible by two with the remainder of 0
+  
+  // a) Create a test with expect statements for each of the variables provided.
+  
+  var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+  // Expected output: [-9, 7, 9, 199]
+  
+  var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+  // Expected output: [-823, 7, 23]
+  
+  describe('onlyOdd', () => {
+      test('return new array of odd numbers sorted from least to greatest', () => {
+        expect(onlyOdd(fullArr1)).toEqual([-9, 7, 9, 199])
+      })
+    })
+  
+//   // b) Create the function that makes the test pass.
+  
+   //declare function named oddNumbers
+   //make variable of new empty array
+   //declare if statement of num divided by two and not equal 0 to get the odd number.
+  const oddNumbers = (sortArr) => {
+      let numOnly = sortArr.filter((value, index) => {
```

Comment:
Great job! I'm writing this as Sarah is going over this problem. Remember you can chain together logic with logical operators. But this is also very readable.